src/main/java/com/google/cloud/genomics/dataflow/readers/bam/Reader.java

```diff
@@ -44,22 +44,38 @@
   BAMShard shard;
   DoFn<BAMShard, Read>.ProcessContext c;
   Stopwatch timer;
-  ValidationStringency stringency;
+  ReaderOptions options;
   
   SAMRecordIterator iterator;
-  boolean procesingUnmapped;
+  enum Filter {
+    UNMAPPED_ONLY,
+    MAPPED_AND_UNMAPPED,
+    MAPPED_ONLY
+  }
+  
+  Filter filter;
   
   int recordsBeforeStart = 0;
   int recordsAfterEnd = 0;
   int mismatchedSequence = 0;
   int recordsProcessed = 0;
   
-  public Reader(Objects storageClient, ValidationStringency stringency, BAMShard shard, DoFn<BAMShard, Read>.ProcessContext c) {
+  public Reader(Objects storageClient, ReaderOptions options, BAMShard shard, DoFn<BAMShard, Read>.ProcessContext c) {
     super();
     this.storageClient = storageClient;
     this.shard = shard;
     this.c = c;
-    this.stringency = stringency;
+    this.options = options;
+    filter = setupFilter(options, shard.contig.referenceName);
+  }
+  
+  public static Filter setupFilter(ReaderOptions options, String referenceName) {
+    if (referenceName.equals("*")) {
```

Comment:
The common way of asking for unmapped reads is by specifying asterisk
     * as the sequence name.
     * From SAM format, section 1.4. paragraph 3:
     * "RNAME: Reference sequence NAME of the alignment. 
     * If @SQ header lines are present, RNAME (if not ‘*’) must be present in 
     * one of the SQ-SN tag. 
     * An unmapped segment without coordinate has a ‘*’ at this field."
     * https://samtools.github.io/hts-specs/SAMv1.pdf
I added the comment.